app/pages/activity/components/emoticonPicker.tsx

```diff
@@ -0,0 +1,68 @@
+"use client";
+import { useState } from "react";
+import { Button } from "@/components/ui/button";
+import { Card, CardContent } from "@/components/ui/card";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { ActivityRequestTypeEnum } from "@/api/models/activity-request";
+import React from "react";
+import { icons } from "../utils/icons";
+import { Smile } from "lucide-react";
+
+interface EmoticonPickerProps {
+  onSelect: (activityType: ActivityRequestTypeEnum) => void;
+}
+
+const EmoticonPicker: React.FC<EmoticonPickerProps> = ({ onSelect }) => {
+  const [selectedEmoticon, setSelectedEmoticon] = useState<string | null>(null);
```

Comment:
tutaj też undefined

Reply:
czemu undefined jest preferowany od null ? bo 
type x = {
 value? : string 
} to jest basic syntax w typescript. 

pytajnik oznacza undefined  i to często jest używane, więc później  będzie  bo w jednej części kodu będą nulle bo je dodałaś a w drugiej undefindy bo pytajnik ?. więc lepiej odrazu wszędzie undefined pisać